docs/migration-v2/index.md

```diff
@@ -0,0 +1,237 @@
+promoting reuse. It also makes some future-proofing changes to the way you
+configure and set up hooks. Let's dive in!
+
+## Configuration
+
+Previously, the `createHooks` function accepted two arguments: the first one
+declaring the hooks, and the second passing configuration options. In v2, these
+have been merged into a single configuration object. The `sort` option has also
+been enabled by default and enhanced to provide more granular control over how
+declarations are prioritized.
+
+### Merging hook declarations and configuration options
+
+Pass a single object to the `createHooks` function. Move hook declarations under
+the `hooks` field, and add any configuration options that were previously passed
+in the second argument.
+
+#### Before
+
+```typescript
+// src/css.ts
+
+export const [hooks, css] = createHooks(
+  {
+    "&:hover": "&:hover",
+  },
+  {
+    fallback: "revert-layer",
+    debug: true,
+  },
+);
+```
+
+#### After
+
+```typescript
+// src/css.ts
+
+export const { styleSheet, css } = createHooks({
+  hooks: {
+    "&:hover": "&:hover",
+  },
+  fallback: "revert-layer",
+  debug: true,
+});
+```
+
+### Updating the `sort` option
+
+If you previously used `sort: true`, you can simply remove that option, as it is
+now enabled by default.
+
+Otherwise, you can add the following to your configuration to disable sorting,
+which most closely resembles the default behavior in v1:
+
+```typescript
+sort: {
+  properties: false,
+  conditionalStyles: false
+}
+```
+
+For more information about sorting, please review the configuration guide.
+
+## Setup
+
+In v1, the `createHooks` function returned a tuple containing (1) a CSS string
+(the style sheet needed to support the configured hooks) and (2) the `css`
+function used to create inline style rules. In v2, it returns an object instead.
+This can be destructured in a similar manner.
+
+### Destructuring the `createHooks` return value
+
+Change the array (tuple) destructuring syntax to use object destructuring syntax
+instead. Note that what was previously called `hooks` (by convention) has been
+renamed to `styleSheet`.
+
+#### Before
+
+```typescript
+// src/css.ts
+
+export const [hooks, css] = createHooks(/* ... */);
+```
+
+#### After
+
+```typescript
+// src/css.ts
+
+export const { styleSheet, css } = createHooks(/* ... */);
+```
+
+### Adding the style sheet
+
+Find where the style sheet is rendered, e.g. in your root component. Update the
+`hooks` import to the new name `styleSheet`, and invoke it as a function to
+render the CSS string.
+
+#### Before
+
+```tsx
+// src/app.tsx
+
+import { hooks } from "./css";
+
+export function App() {
+  return (
+    <>
+      <style dangerouslySetInnerHTML={{ __html: hooks }} />
+      <HomePage />
+    </>
+  );
+}
+```
+
+#### After
+
+```tsx
+// src/app.tsx
+
+import { styleSheet } from "./css";
+
+export function App() {
+  return (
+    <>
+      <style dangerouslySetInnerHTML={{ __html: styleSheet() }} />
```

Comment:
I'm curious if `createHooks` can return a component directly that renders this — are there cases where they might be passing in unsanitized data so the callsite should remind them it's unsafe?

Reply:
The only issue that I can think of would be the direct framework dependency, but honestly that's probably not much of a downside considering that there separate packages tailored to each framework. Good idea!